base/path.jl

```diff
@@ -338,28 +338,6 @@ abspath(a::AbstractString, b::AbstractString...) = abspath(joinpath(a,b...))
 
 if Sys.iswindows()
 
-function realpath(path::AbstractString)
-    h = ccall(:CreateFileW, stdcall, Int, (Cwstring, UInt32, UInt32, Ptr{Cvoid}, UInt32, UInt32, Int),
-                path, 0, 0x03, C_NULL, 3, 0x02000000, 0)
-    windowserror(:realpath, h == -1)
-    try
-        buf = Vector{UInt16}(undef, 256)
-        oldlen = len = length(buf)
-        while len >= oldlen
-            len = ccall(:GetFinalPathNameByHandleW, stdcall, UInt32, (Int, Ptr{UInt16}, UInt32, UInt32),
-                            h, buf, (oldlen=len)-1, 0x0)
-            windowserror(:realpath, iszero(len))
-            resize!(buf, len) # strips NUL terminator on last call
-        end
-        if 4 < len < 264 && 0x005c == buf[1] == buf[2] == buf[4] && 0x003f == buf[3]
```

Comment:
I seem to recall that libuv is missing the length check, but otherwise lgtm

Reply:
You mean a max path type length check? I don't think I see that this is an issue.

Reply:
libuv [doesn't check that `len > 4`](https://github.com/libuv/libuv/blob/813264add0c3b5fba782e4bb849bffad4701ecf6/src/win/fs.c#L2496-L2498), but since it uses `wcsncmp` (which stops at NUL chars) it should be fine there.  It doesn't check `len < 264` either, but I guess excluding actual paths with length > 263 that start with `\\?\` is less of a practical concern?

Reply:
Paths that start with `\\?\` are extended paths and  can extend up to 32 767 characters